addons/html_editor/static/src/main/link/link_popover.js

```diff
@@ -550,4 +567,37 @@ export class LinkPopover extends Component {
     isAttachmentUrl() {
         return !!this.state.url.match(/\/web\/content\/\d+/);
     }
+    /**
+     * Checks if the given URL is using the domain where the content being
+     * edited is reachable, i.e. if this URL should be stripped of its domain
+     * part and converted to a relative URL if put as a link in the content.
+     *
+     * @private
+     * @returns {boolean}
+     */
+    isAbsoluteURLInCurrentDomain() {
+        // First check if it is a relative URL: if it is, we don't want to check
+        // further as we will always leave those untouched.
+        let hasProtocol;
+        try {
+            hasProtocol = !!new URL(this.state.url).protocol;
+        } catch {
+            hasProtocol = false;
+        }
+        if (!hasProtocol) {
+            return false;
+        }
+
+        const urlObj = new URL(this.state.url, window.location.origin);
+        // Chosen heuristic to detect someone trying to enter a link using
+        // its Odoo instance domain. We just suppose it should be a relative
+        // URL (if unexpected behavior, the user can just not enter its Odoo
+        // instance domain but its real domain, or opt-out from the domain
+        // stripping). Mentioning an .odoo.com domain, especially its own
+        // one, is always a bad practice anyway.
+        return (
+            urlObj.origin === window.location.origin ||
+            new RegExp(`^https?://${session.db}\\.odoo\\.com(/.*)?$`).test(urlObj.origin)
```

Comment:
Okkk, so this regex is to check the url when the user inputs the odoo instance domain (db.odoo.com) and then consider it as relative url, right? I didn't aware that we could also have this case.

Do you think we should apply similar checking when we retrieve internal url preview (currently we only check the host names are matching)? (we will do it in another task/pr ofc, if you think it's more complete/better) :pray:

Reply:
> Do you think we should apply similar checking when we retrieve internal url preview

@Jinjiu96 what do you mean? If it's just a preview, it does not matter no? But not sure what you are talking about precisely 

The point of this code is to create a "proper" "website link" which, most of the time, is using a relative URL, independently from what is pasted and where the user is editing the website from. Detailed explanation in the commit message and in the code here: https://github.com/odoo/odoo/commit/0961056f2b53267a61ca868153a191ff3c38cad2

Reply:
Thanksss, I will create a task in our pipe :pray: